src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,187 @@
+package duke;
+
+import duke.command.AddDeadlineCommand;
+import duke.command.AddEventCommand;
+import duke.command.AddToDoCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+
+/**
+ * Encapsulates the unit that parses user inputs.
+ */
+
+public class Parser {
+    /**
+     * Parses a user input command as a Command object.
+     *
+     * @param fullCommand String containing a command that is input by user.
+     * @return A Command object that corresponds to fullCommand.
+     * @throws DukeException If fullCommand is not properly formatted.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
```

Comment:
Well done on this method. It is really well organized and its easy for the reader to see your code. Keep it up!